cluster/addons/addon-manager/kube-addons.sh

```diff
@@ -106,7 +106,9 @@ function log() {
 
 # Generate kubectl prune-whitelist flags from provided resource list.
 function generate_prune_whitelist_flags() {
-  local -r resources=($@)
+  local resources
+  IFS=" " read -r -a resources <<< "$@"
+  readonly resources
```

Comment:
change reason:
```
In ./cluster/addons/addon-manager/kube-addons.sh line 109:
  local -r resources=($@)
                      ^-- SC2206: Quote to prevent word splitting, or split robustly with mapfile or read -a.
```

Reply:
The suggested change is simpler and more idiomatic.  Also it is correct, while this is not (if I understand):

```
$ function foo() { local resources=("$@"); declare -p resources; }; foo a b "c d e"
declare -a resources=([0]="a" [1]="b" [2]="c d e")
```

That seems right.

```
$ function bar() { local resources; IFS=" " read -r -a resources <<< "$@"; declare -p resources; }; bar a b "c d e"
declare -a resources=([0]="a" [1]="b" [2]="c" [3]="d" [4]="e")
```

That seems wrong.